### DISCUSSION TITLE
The write function within the "useContractWrite" code is not functioning properly on mobile devices

### DISCUSSION THREAD

**Discussion by lucaslokchan** [wevm/wagmi#2924] [Q&A]
The code snippet functions correctly when the button is clicked on a desktop, as it triggers the expected behavior of a Metamask plugin popping up for a transaction. However, when the same button is clicked on a mobile device using browsers such as Metamask in-app browser, Brave, or Chrome on iOS, no action occurs. Is there a solution available to resolve this issue, allowing the button to redirect users to the Metamask app for transaction confirmation?

```
  const { data, isLoading, isSuccess, write } = useContractWrite({
    mode: "recklesslyUnprepared",
    address: contractAddress,
    abi: abi,
    functionName: functionName,
    chainId: sepolia.chainId,
    args: args,
   });
  
  return(
   <Button
      onClick={() =>
      write?.()
     }
   >
  )
```

**Comment by branzinho** [comment]
Hi Lucas, what was the solution for this issue?
